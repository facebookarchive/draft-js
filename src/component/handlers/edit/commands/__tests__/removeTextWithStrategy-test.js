/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall draft_js
 */

'use strict';
import type ContentState from 'ContentState';

const BlockMapBuilder = require('BlockMapBuilder');
const ContentBlockNode = require('ContentBlockNode');
const EditorState = require('EditorState');
const SelectionState = require('SelectionState');
const UnicodeUtils = require('UnicodeUtils');

const getSampleStateForTesting = require('getSampleStateForTesting');
const Immutable = require('immutable');
const moveSelectionForward = require('moveSelectionForward');
const removeTextWithStrategy = require('removeTextWithStrategy');

jest.mock('generateRandomKey');
const toggleExperimentalTreeDataSupport = (enabled: boolean) => {
  jest.doMock('gkx', () => name => {
    return name === 'draft_tree_data_support' ? enabled : false;
  });
};
// Seems to be important to put this at the top
toggleExperimentalTreeDataSupport(true);

const {List} = Immutable;

const {contentState} = getSampleStateForTesting();

const contentBlockNodes = [
  new ContentBlockNode({
    key: 'A',
    nextSibling: 'B',
    text: 'Alpha',
    type: 'blockquote',
  }),
  new ContentBlockNode({
    key: 'B',
    prevSibling: 'A',
    nextSibling: 'G',
    type: 'ordered-list-item',
    children: List(['C', 'F']),
  }),
  new ContentBlockNode({
    parent: 'B',
    key: 'C',
    nextSibling: 'F',
    type: 'blockquote',
    children: List(['D', 'E']),
  }),
  new ContentBlockNode({
    parent: 'C',
    key: 'D',
    nextSibling: 'E',
    type: 'header-two',
    text: 'Delta',
  }),
  new ContentBlockNode({
    parent: 'C',
    key: 'E',
    prevSibling: 'D',
    type: 'unstyled',
    text: 'Elephant',
  }),
  new ContentBlockNode({
    parent: 'B',
    key: 'F',
    prevSibling: 'C',
    type: 'code-block',
    text: 'Fire',
  }),
  new ContentBlockNode({
    key: 'G',
    prevSibling: 'B',
    nextSibling: 'H',
    type: 'ordered-list-item',
    text: 'Gorila',
  }),
  new ContentBlockNode({
    key: 'H',
    prevSibling: 'G',
    nextSibling: 'I',
    text: ' ',
    type: 'atomic',
  }),
  new ContentBlockNode({
    key: 'I',
    prevSibling: 'H',
    text: 'last',
    type: 'unstyled',
  }),
];

const assertRemoveTextOperation = (
  operation: (editorState: EditorState) => ContentState,
  selection:
    | $TEMPORARY$object<{...}>
    | $TEMPORARY$object<{
        anchorKey: $TEMPORARY$string<'D'>,
        anchorOffset: number,
        focusKey: $TEMPORARY$string<'D'>,
        focusOffset: number,
      }>
    | $TEMPORARY$object<{
        anchorKey: $TEMPORARY$string<'D'>,
        anchorOffset: number,
        focusKey: $TEMPORARY$string<'E'>,
        focusOffset: number,
      }>
    | $TEMPORARY$object<{
        anchorKey: $TEMPORARY$string<'E'>,
        anchorOffset: number,
        focusKey: $TEMPORARY$string<'E'>,
        focusOffset: number,
      }> = {},
  content: $TEMPORARY$array<ContentBlockNode> = contentBlockNodes,
) => {
  const result = operation(
    EditorState.forceSelection(
      EditorState.createWithContent(
        contentState.setBlockMap(BlockMapBuilder.createFromArray(content)),
      ),
      SelectionState.createEmpty(content[0].key).merge(selection),
    ),
  );
  const expected = result.getBlockMap().toJS();

  expect(expected).toMatchSnapshot();
};

test(`at end of a leaf block and sibling is another leaf block forward delete concatenates`, () => {
  assertRemoveTextOperation(
    editorState =>
      removeTextWithStrategy(
        editorState,
        strategyState => {
          const selection = strategyState.getSelection();
          const content = strategyState.getCurrentContent();
          const key = selection.getAnchorKey();
          const offset = selection.getAnchorOffset();
          const charAhead = content.getBlockForKey(key).getText()[offset];
          return moveSelectionForward(
            strategyState,
            charAhead ? UnicodeUtils.getUTF16Length(charAhead, 0) : 1,
          );
        },
        'forward',
      ),
    {
      anchorKey: 'D',
      anchorOffset: contentBlockNodes[3].getLength(),
      focusKey: 'D',
      focusOffset: contentBlockNodes[3].getLength(),
    },
  );
});

test(`at end of a leaf block and sibling is not another leaf block forward delete is no-op`, () => {
  // no next sibling
  assertRemoveTextOperation(
    editorState =>
      removeTextWithStrategy(
        editorState,
        strategyState => {
          const selection = strategyState.getSelection();
          const content = strategyState.getCurrentContent();
          const key = selection.getAnchorKey();
          const offset = selection.getAnchorOffset();
          const charAhead = content.getBlockForKey(key).getText()[offset];
          return moveSelectionForward(
            strategyState,
            charAhead ? UnicodeUtils.getUTF16Length(charAhead, 0) : 1,
          );
        },
        'forward',
      ),
    {
      anchorKey: 'E',
      anchorOffset: contentBlockNodes[4].getLength(),
      focusKey: 'E',
      focusOffset: contentBlockNodes[4].getLength(),
    },
  );
  // next sibling is not a leaf
  assertRemoveTextOperation(
    editorState =>
      removeTextWithStrategy(
        editorState,
        strategyState => {
          const selection = strategyState.getSelection();
          const content = strategyState.getCurrentContent();
          const key = selection.getAnchorKey();
          const offset = selection.getAnchorOffset();
          const charAhead = content.getBlockForKey(key).getText()[offset];
          return moveSelectionForward(
            strategyState,
            charAhead ? UnicodeUtils.getUTF16Length(charAhead, 0) : 1,
          );
        },
        'forward',
      ),
    {
      anchorKey: 'E',
      anchorOffset: contentBlockNodes[4].getLength(),
      focusKey: 'E',
      focusOffset: contentBlockNodes[4].getLength(),
    },
  );
});

test(`across blocks with forward delete is a no-op`, () => {
  assertRemoveTextOperation(
    editorState =>
      removeTextWithStrategy(
        editorState,
        strategyState => {
          const selection = strategyState.getSelection();
          const content = strategyState.getCurrentContent();
          const key = selection.getAnchorKey();
          const offset = selection.getAnchorOffset();
          const charAhead = content.getBlockForKey(key).getText()[offset];
          return moveSelectionForward(
            strategyState,
            charAhead ? UnicodeUtils.getUTF16Length(charAhead, 0) : 1,
          );
        },
        'forward',
      ),
    {
      anchorKey: 'D',
      anchorOffset: contentBlockNodes[3].getLength(),
      focusKey: 'E',
      focusOffset: contentBlockNodes[4].getLength(),
    },
  );
});
