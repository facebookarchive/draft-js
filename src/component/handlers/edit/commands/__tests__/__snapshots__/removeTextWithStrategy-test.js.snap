// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`across blocks with forward delete is a no-op 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "Alpha",
    "type": "blockquote",
  },
  "B": Object {
    "characterList": Array [],
    "children": Array [
      "C",
      "F",
    ],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "G",
    "parent": null,
    "prevSibling": "A",
    "text": "",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [],
    "children": Array [
      "D",
      "E",
    ],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": "F",
    "parent": "B",
    "prevSibling": null,
    "text": "",
    "type": "blockquote",
  },
  "D": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "D",
    "nextSibling": "E",
    "parent": "C",
    "prevSibling": null,
    "text": "Delta",
    "type": "header-two",
  },
  "E": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "E",
    "nextSibling": null,
    "parent": "C",
    "prevSibling": "D",
    "text": "Elephant",
    "type": "unstyled",
  },
  "F": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "F",
    "nextSibling": null,
    "parent": "B",
    "prevSibling": "C",
    "text": "Fire",
    "type": "code-block",
  },
  "G": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "G",
    "nextSibling": "H",
    "parent": null,
    "prevSibling": "B",
    "text": "Gorila",
    "type": "ordered-list-item",
  },
  "H": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "H",
    "nextSibling": "I",
    "parent": null,
    "prevSibling": "G",
    "text": " ",
    "type": "atomic",
  },
  "I": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "I",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "H",
    "text": "last",
    "type": "unstyled",
  },
}
`;

exports[`at end of a leaf block and sibling is another leaf block forward delete concatenates 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "Alpha",
    "type": "blockquote",
  },
  "B": Object {
    "characterList": Array [],
    "children": Array [
      "C",
      "F",
    ],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "G",
    "parent": null,
    "prevSibling": "A",
    "text": "",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [],
    "children": Array [
      "D",
    ],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": "F",
    "parent": "B",
    "prevSibling": null,
    "text": "",
    "type": "blockquote",
  },
  "D": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "D",
    "nextSibling": null,
    "parent": "C",
    "prevSibling": null,
    "text": "DeltaElephant",
    "type": "header-two",
  },
  "F": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "F",
    "nextSibling": null,
    "parent": "B",
    "prevSibling": "C",
    "text": "Fire",
    "type": "code-block",
  },
  "G": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "G",
    "nextSibling": "H",
    "parent": null,
    "prevSibling": "B",
    "text": "Gorila",
    "type": "ordered-list-item",
  },
  "H": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "H",
    "nextSibling": "I",
    "parent": null,
    "prevSibling": "G",
    "text": " ",
    "type": "atomic",
  },
  "I": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "I",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "H",
    "text": "last",
    "type": "unstyled",
  },
}
`;

exports[`at end of a leaf block and sibling is not another leaf block forward delete is no-op 1`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "Alpha",
    "type": "blockquote",
  },
  "B": Object {
    "characterList": Array [],
    "children": Array [
      "C",
      "F",
    ],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "G",
    "parent": null,
    "prevSibling": "A",
    "text": "",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [],
    "children": Array [
      "D",
      "E",
    ],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": "F",
    "parent": "B",
    "prevSibling": null,
    "text": "",
    "type": "blockquote",
  },
  "D": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "D",
    "nextSibling": "E",
    "parent": "C",
    "prevSibling": null,
    "text": "Delta",
    "type": "header-two",
  },
  "E": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "E",
    "nextSibling": null,
    "parent": "C",
    "prevSibling": "D",
    "text": "Elephant",
    "type": "unstyled",
  },
  "F": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "F",
    "nextSibling": null,
    "parent": "B",
    "prevSibling": "C",
    "text": "Fire",
    "type": "code-block",
  },
  "G": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "G",
    "nextSibling": "H",
    "parent": null,
    "prevSibling": "B",
    "text": "Gorila",
    "type": "ordered-list-item",
  },
  "H": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "H",
    "nextSibling": "I",
    "parent": null,
    "prevSibling": "G",
    "text": " ",
    "type": "atomic",
  },
  "I": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "I",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "H",
    "text": "last",
    "type": "unstyled",
  },
}
`;

exports[`at end of a leaf block and sibling is not another leaf block forward delete is no-op 2`] = `
Object {
  "A": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "A",
    "nextSibling": "B",
    "parent": null,
    "prevSibling": null,
    "text": "Alpha",
    "type": "blockquote",
  },
  "B": Object {
    "characterList": Array [],
    "children": Array [
      "C",
      "F",
    ],
    "data": Object {},
    "depth": 0,
    "key": "B",
    "nextSibling": "G",
    "parent": null,
    "prevSibling": "A",
    "text": "",
    "type": "ordered-list-item",
  },
  "C": Object {
    "characterList": Array [],
    "children": Array [
      "D",
      "E",
    ],
    "data": Object {},
    "depth": 0,
    "key": "C",
    "nextSibling": "F",
    "parent": "B",
    "prevSibling": null,
    "text": "",
    "type": "blockquote",
  },
  "D": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "D",
    "nextSibling": "E",
    "parent": "C",
    "prevSibling": null,
    "text": "Delta",
    "type": "header-two",
  },
  "E": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "E",
    "nextSibling": null,
    "parent": "C",
    "prevSibling": "D",
    "text": "Elephant",
    "type": "unstyled",
  },
  "F": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "F",
    "nextSibling": null,
    "parent": "B",
    "prevSibling": "C",
    "text": "Fire",
    "type": "code-block",
  },
  "G": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "G",
    "nextSibling": "H",
    "parent": null,
    "prevSibling": "B",
    "text": "Gorila",
    "type": "ordered-list-item",
  },
  "H": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "H",
    "nextSibling": "I",
    "parent": null,
    "prevSibling": "G",
    "text": " ",
    "type": "atomic",
  },
  "I": Object {
    "characterList": Array [
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
      Object {
        "entity": null,
        "style": Array [],
      },
    ],
    "children": Array [],
    "data": Object {},
    "depth": 0,
    "key": "I",
    "nextSibling": null,
    "parent": null,
    "prevSibling": "H",
    "text": "last",
    "type": "unstyled",
  },
}
`;
